RecoLocalTracker/SiPixelRecHits/interface/PixelCPEBase.h

```diff
@@ -43,12 +43,50 @@
 #include "CondFormats/SiPixelObjects/interface/SiPixelTemplateDBObject.h"
 
 #include <unordered_map>
+#include "tbb/concurrent_unordered_map.h"
 
 #include <iostream>
 #ifdef EDM_ML_DEBUG
 #include <atomic>
 #endif
 
+class PixelCPEWarningSummary {
+private:
+  bool m_debug;
+  std::string m_category;
+  tbb::concurrent_unordered_map<std::string, std::atomic<std::size_t>> m_warnings;
+
+public:
+  PixelCPEWarningSummary(const std::string& category, bool debug = false) : m_debug(debug), m_category(category) {}
+
+  void add(const std::string& message, const std::string& details) {
```

Comment:
considering the  frequency of calls to `PixelCPEBase`, I think that for the production code the key should be changed to something less open and hardcoded, perhaps just an enum.
In that case the map can be changed to just an array.